vignettes/datatable-importing.Rmd

```diff
@@ -167,7 +169,9 @@ my.write = function (x) {
 }
 ```
 
-When using a package as a suggested dependency, you should not `import` it in the `NAMESPACE` file. Just mention it in the `DESCRIPTION` file. You also _must_ use the `data.table::` prefix when calling `data.table` functions because none of them are imported.
+When using a package as a suggested dependency, you should not `import` it in the `NAMESPACE` file. Just mention it in the `DESCRIPTION` file. 
+When using `data.table` functions in package code (R/* files) you need to use the `data.table::` prefix because none of them are imported. 
+When using `data.table` in package tests (e.g. tests/testthat/test* files), you need to declare `.datatable.aware=TRUE` in one of the R/* files.
```

Comment:
same here, good to know if presence of NAMESPACE file matters

Reply:
I don't think NAMESPACE matters. presence of `.datatable.aware=TRUE` means no error on `testthat::test_package` (as expected); absence means unexpected "object not found" error.

Reply:
I think all this can be answered with `cedta` code:

```
isTRUE(ns$.datatable.aware)
```

where `ns` is the namespace of the call, so `.datatable.aware` covers all bases.

Earlier we see:

```
"data.table" %chin% names(getNamespaceImports(ns))
# getNamespaceImports main code:
.getNamespaceInfo(ns, "imports")
```

AFAICT these `names` correspond to `importFrom` or `import` statements in `NAMESPACE`

Reply:
not sure I understand. can you please clarify? are you saying that you may be able to modify cedta to fix the issue, without having me add `.datatable.aware=TRUE` in my package? that would be ideal.

Reply:
no, I think without being in imports, `.datatable.aware` is probably the canonical way to do things. `.datatable.aware=TRUE` seems pretty minimal after all! Just a quick line of code to signal to `data.table` that you know we're here.

My comment was hoping to shed light on the discussion on whether `NAMESPACE` matters

Reply:
ok